restcomm/restcomm.extension.controller/src/main/java/org/restcomm/connect/extension/controller/ExtensionController.java

```diff
@@ -116,4 +117,55 @@ public ExtensionResponse executePostOutboundAction(final IExtensionRequest er, L
         //TODO: implement actual calls
         return response;
     }
+
+    public ExtensionResponse executePreInboundAction(final IExtensionRequest er, List<RestcommExtensionGeneric> extensions) {
+        ExtensionResponse response = new ExtensionResponse();
+//        response.setAllowed(true);
+        if (extensions != null && extensions.size() > 0) {
+            for (RestcommExtensionGeneric extension : extensions) {
+                if(logger.isInfoEnabled()) {
+                    logger.info( extension.getName()+" is enabled="+extension.isEnabled());
+                }
+                if (extension.isEnabled()) {
+                    response = extension.preInboundAction(er);
+                    //fail fast
+                    if (!response.isAllowed()){
+                        break;
+                    }
+                }
+            }
+        }
+        return response;
+    }
+
+    public ExtensionResponse executePostInboundAction(final IExtensionRequest er,  List<RestcommExtensionGeneric> extensions) {
```

Comment:
this extension API is starting to feel very specific ,with many variants of same purpose. Not sure if we could push all this executePre/PostXAction methods into the request interface, so the request tells the type of exec point X by a using a getter in the req interface... I think we could have a single pre and post methods...